php-frontend/src/main/java/org/sonar/php/tree/symbols/DeclarationVisitor.java

```diff
@@ -103,6 +104,10 @@ public void visitCompilationUnit(CompilationUnitTree tree) {
     classSymbolDataByTree.forEach((declaration, symbolData) -> {
       ClassSymbol symbol = classSymbolIndex.get(symbolData);
       ((ClassDeclarationTreeImpl) declaration).setSymbol(symbol);
+      if (methodsByClassTree.containsKey(declaration)) {
+        methodsByClassTree.get(declaration)
+          .forEach(methodData -> methodTreeByData.get(methodData).setSymbol(symbol.getDeclaredMethod(methodData.name())));
+      }
```

Comment:
This commit doesn't change the unit tests: I believe we should have a unit test which checks that the symbol associated to the MethodDeclarationTree is the same object as the method symbol referenced by the class symbol.